### PR TITLE
[WIP][Don't merge] remove pickle in gluon dataloader

### PR DESCRIPTION
**Pull request by wkcn**
## Description ##
It seems that `pickle` is not necessary in gluon dataloder.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: python/mxnet/gluon/data/dataloader.py [hunk 1]
```diff
@@ -398,9 +398,7 @@ def _worker_fn(samples, batchify_fn, dataset=None):
     # preserving dataset as global variable can save tons of overhead and is safe in new process
     global _worker_dataset
     batch = batchify_fn([_worker_dataset[i] for i in samples])
-    buf = io.BytesIO()
-    ForkingPickler(buf, pickle.HIGHEST_PROTOCOL).dump(batch)
-    return buf.getvalue()
+    return batch
```
Comment (zhreshold) [inline python/mxnet/gluon/data/dataloader.py]: pickle's protocol isn't quite consistent across versions.
this may or may not be a problem, however I don't have any benchmark result to support either one.

Comment (wkcn) [reply python/mxnet/gluon/data/dataloader.py]: Yes. I am working on it. I found the performance of PyTorch is faster than that of MXNet, and I am looking for where the difference is.

**Comment by wkcn** [comment]
Close it since I found the bottleneck of performance is the release of NDArray Object.
